e2etest/l2tests/interface_selector.go

```diff
@@ -173,5 +177,212 @@ var _ = ginkgo.Describe("L2-interface selector", func() {
+							},
+							Spec: metallbv1beta1.L2AdvertisementSpec{
+								Interfaces:    []string{NodeNics[0]},
+								NodeSelectors: k8s.SelectorsForNodes([]v1.Node{node}),
+							},
+						},
+					},
+				}
+
+				err = ConfigUpdater.Update(resources)
+				framework.ExpectNoError(err)
+
+				ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
+				err := mac.FlushIPNeigh(ingressIP, executor.Host)
+				framework.ExpectNoError(err)
+
+				gomega.Eventually(func() string {
+					err := mac.RequestAddressResolutionFromIface(ingressIP, LocalNics[0], executor.Host)
+					if err != nil {
+						return err.Error()
+					}
+					err = service.ValidateL2(svc)
+					if err != nil {
+						return err.Error()
+					}
+
+					advNode, err := advertisingNodeFromMAC(allNodes.Items, ingressIP, executor.Host)
+					if err != nil {
+						return err.Error()
+					}
+
+					return advNode.Name
+				}, 1*time.Minute, 1*time.Second).Should(gomega.Equal(node.Name))
+			}
+		})
+	})
+
+	ginkgo.Context("Modify Address pool", func() {
+		ginkgo.BeforeEach(func() {
+			resources := internalconfig.ClusterResources{
+				L2Advs: []metallbv1beta1.L2Advertisement{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "with-interfaces",
+						},
+						Spec: metallbv1beta1.L2AdvertisementSpec{
+							Interfaces: []string{NodeNics[0]},
+						},
+					},
+				},
+			}
+
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("Modify Address pool", func() {
+			startIPV4, _ := config.GetIPFromRangeByIndex(IPV4ServiceRange, 0)
+			endIPV4, _ := config.GetIPFromRangeByIndex(IPV4ServiceRange, 1)
+			startIPV6, _ := config.GetIPFromRangeByIndex(IPV6ServiceRange, 0)
+			endIPV6, _ := config.GetIPFromRangeByIndex(IPV6ServiceRange, 1)
+
+			resources := internalconfig.ClusterResources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "l2-test",
+						},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{
+								startIPV4 + "-" + startIPV4,
+								startIPV6 + "-" + startIPV6,
+							},
+						},
+					},
+				},
+			}
+
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			svc, _ := service.CreateWithBackend(cs, f.Namespace.Name, fmt.Sprintf("lb-service"))
+			defer func() {
+				err = cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+			}()
+			framework.ExpectNoError(err)
+
+			ingressIP := e2eservice.GetIngressPoint(&svc.Status.LoadBalancer.Ingress[0])
+			err = mac.FlushIPNeigh(ingressIP, executor.Host)
+			framework.ExpectNoError(err)
+
+			gomega.Eventually(func() error {
+				return mac.RequestAddressResolutionFromIface(ingressIP, LocalNics[0], executor.Host)
+			}, 10*time.Second, 1*time.Second).ShouldNot(gomega.HaveOccurred())
+
+			resources.Pools = []metallbv1beta1.IPAddressPool{
+				{
+					ObjectMeta: metav1.ObjectMeta{
```

Comment:
What's the difference between this and the previous version? I still struggle to understand what this test is trying to do.
One possible good update test is one that changes the set of interfaces, like we move from all the interfaces to a subset and we verify that the change was accepted correctly.

Reply:
This test is trying to test this case: 
we have a address pool connected with the L2Avd which has Interfaces,  then we create a LB service. we expanded the address pool. we check if it can work correctly just like before.
if you think it's not work or not correct, we can delete this test. I will add more test in next PR.